spec/serializers/search_and_compare/course_serializer_spec.rb

```diff
@@ -98,16 +148,25 @@
         HashWithIndifferentAccess.new(JSON.parse(file))
       end
 
-      describe 'Provider_serializer_Mapping' do
+      # This test is the overall test of 'rails' vs 'csharp' generated json
+      context 'orginal json' do
+        it { should eq expected_json }
+      end
+
+      # All the tests belows are the more intimate for expected 32 fields and its related nested fields
+      # Tests consistence of
+      #   database models mapping to json
+      #   default values
+      context 'Provider_serializer_Mapping' do
         # testing the provider serializer, its part of the json
         describe 'Provider' do
           subject { resource[:Provider] }
```

Comment:
We should probably not test the provider serialization here. I think we should just test that the generated json from `resource[:Provider]` matches what the provider serializer outputs, you can find docs on how to serialize manually here: https://github.com/jaredbeck/active_model_serializers